python/ray/data/_internal/datasource/parquet_datasource.py

```diff
@@ -230,6 +230,14 @@ def __init__(
         # duplicating the partition data, we disable PyArrow's partitioning.
         dataset_kwargs["partitioning"] = None
 
+        # `read_schema` is the schema object that will be used to perform
+        # read operations.
+        # It should be None, unless user has specified the schema or columns.
+        # We don't use the inferred schema for read, because the pyarrow only infers
+        # schema based on the first file. Thus, files with different schemas will end
```

Comment:
does that mean this is a pyarrow bug?

Reply:
not sure it's a bug, oh they intended to do so to avoid high overheads